app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -429,6 +421,52 @@ class CiYaml {
     _checkExceptions(exceptions);
   }
 
+  void _verifyTargetClassicRepo(pb.Target target, List<String> exceptions) {
+    // These serve mostly as documentation.
+    assert(!isFusion, 'Expected to be called in a non-monorepo');
+    assert(target.runIf.isNotEmpty, 'Expected to be called when non-empty');
+
+    // 1. Every target must depend on .ci.yaml at the root of the repo.
+    if (!target.runIf.contains('.ci.yaml')) {
+      exceptions.add('ERROR: ${target.name} is missing `.ci.yaml` in runIf');
+    }
+
+    // 2. The engine repo must additionally depend on DEPS.
+    if (slug == Config.engineSlug && !target.runIf.contains('DEPS')) {
+      exceptions.add('ERROR: ${target.name} is missing `DEPS` in runIf');
+    }
+  }
+
+  void _verifyTargetFusionRepo(pb.Target target, List<String> exceptions) {
+    // These serve mostly as documentation.
+    assert(isFusion, 'Expected to be called in a fusion monorepo');
+    assert(target.runIf.isNotEmpty, 'Expected to be called when non-empty');
+    assert(slug == Config.flutterSlug, 'Expected to be in the combined repo');
+
+    // 1. Every target must depend on .ci.yaml.
+    // The path depends on whether the framework or engine are being validated;
+    // while both belong in the same (mono)repo, they have separate .ci.yaml
+    // files located in different paths.
+    final ciYamlPath = switch (type) {
+      CiType.fusionEngine => 'engine/src/flutter/.ci.yaml',
+      _ => '.ci.yaml',
+    };
```

Comment:
Can these be added automatically by the `.ci.yaml` parser (or by the logic that evaluates `runIf` conditions)? This way we don't have to specify anything in the yaml files. Seems like they add a bunch of noise. Totally fine we want to take this one step at a time though.

Reply:
Filed as https://github.com/flutter/flutter/issues/160874